packages/lab/src/pill/PillNext.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  forwardRef,
+  MouseEvent,
+  PropsWithChildren,
+  HTMLAttributes,
+  KeyboardEvent,
+} from "react";
+import { useWindow } from "@salt-ds/window";
+import { CloseSmallIcon } from "@salt-ds/icons";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { makePrefixer, useButton } from "@salt-ds/core";
+import pillCss from "./PillNext.css";
+import clsx from "clsx";
+
+type ClickEvent = MouseEvent<Element, globalThis.MouseEvent>;
+
+interface PillProps extends HTMLAttributes<HTMLDivElement> {
+  disabled?: boolean;
+  className?: string;
+  icon?: React.ReactNode;
+  onClose?: (e: ClickEvent | KeyboardEvent<HTMLDivElement>) => void;
+}
+
+const withBaseName = makePrefixer("saltPill");
+
+export const Pill = forwardRef<HTMLDivElement, PropsWithChildren<PillProps>>(
+  function Pill(
+    { onClose, onClick, children, className, icon, ...restProps },
+    ref
+  ) {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-pill",
+      css: pillCss,
+      window: targetWindow,
+    });
+
+    const clickable = onClick !== undefined;
+    const closable = onClose !== undefined;
+    const interactive = clickable || closable;
+
+    if (interactive) {
+      return (
+        <InteractivePill
+          onClose={onClose}
+          onClick={onClick}
+          className={className}
+          icon={icon}
+          {...restProps}
+        >
+          {children}
+        </InteractivePill>
+      );
+    }
+
+    return (
+      <div ref={ref} {...restProps} className={clsx(withBaseName(), className)}>
+        {icon}
+        <span className={withBaseName("label")}>{children}</span>
+      </div>
+    );
+  }
+);
+
+const InteractivePill = forwardRef<
```

Comment:
this and the close button should go in another file

Reply:
For sure 👍